src/wasm/wasm-binary.cpp

```diff
@@ -1549,8 +1549,28 @@ void WasmBinaryWriter::trackExpressionDelimiter(Expression* curr,
 }
 
 std::optional<BufferWithRandomAccess> WasmBinaryWriter::writeCodeAnnotations() {
-  // Assemble the info for Branch Hinting: for each function, a vector of the
-  // hints.
+  std::optional<BufferWithRandomAccess> ret;
+
+  auto append = [&](std::optional<BufferWithRandomAccess>&& temp) {
```

Comment:
Why make a lambda just to call it once afterward? Because there will be more calls in the future?

Reply:
Yes, each hint will call append, so we'll have a list of branch hints, inline hints, etc